src/panels/config/integrations/integration-panels/zha/dialog-zha-manage-zigbee-device.ts

```diff
@@ -0,0 +1,303 @@
+import {
+  css,
+  CSSResultGroup,
+  html,
+  LitElement,
+  PropertyValues,
+  TemplateResult,
+} from "lit";
+import { mdiClose } from "@mdi/js";
+import { customElement, property, state } from "lit/decorators";
+import { cache } from "lit/directives/cache";
+import { fireEvent } from "../../../../../common/dom/fire_event";
+import "../../../../../components/ha-code-editor";
+import { createCloseHeading } from "../../../../../components/ha-dialog";
+import {
+  fetchBindableDevices,
+  fetchGroups,
+  ZHADevice,
+  ZHAGroup,
+} from "../../../../../data/zha";
+import { haStyleDialog } from "../../../../../resources/styles";
+import { HomeAssistant } from "../../../../../types";
+import { sortZHADevices, sortZHAGroups } from "./functions";
+import "./zha-cluster-attributes";
+import "./zha-cluster-commands";
+import "./zha-manage-clusters";
+import "./zha-device-binding";
+import "./zha-group-binding";
+import "./zha-device-children";
+import "./zha-device-signature";
+import {
+  Tab,
+  ZHAManageZigbeeDeviceDialogParams,
+} from "./show-dialog-zha-manage-zigbee-device";
+
+@customElement("dialog-zha-manage-zigbee-device")
+class DialogZHAManageZigbeeDevice extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ type: Boolean, reflect: true }) public large = false;
+
+  @state() private _currTab: Tab = "clusters";
+
+  @state() private _device?: ZHADevice;
+
+  @state() private _bindableDevices: ZHADevice[] = [];
+
+  @state() private _groups: ZHAGroup[] = [];
+
+  public async showDialog(
+    params: ZHAManageZigbeeDeviceDialogParams
+  ): Promise<void> {
+    this._device = params.device;
+    if (!this._device) {
+      this.closeDialog();
+      return;
+    }
+    this._currTab = params.tab || "clusters";
+    this.large = false;
+  }
+
+  public closeDialog() {
+    this._device = undefined;
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  protected firstUpdated(changedProps: PropertyValues) {
+    super.firstUpdated(changedProps);
+    this.addEventListener("close-dialog", () => this.closeDialog());
+  }
+
+  protected willUpdate(changedProps: PropertyValues) {
+    super.willUpdate(changedProps);
+    if (!this._device) {
+      return;
+    }
+    const tabs = this._getTabs();
+    if (!tabs.includes(this._currTab)) {
+      this._currTab = tabs[0];
+    }
```

Comment:
do we need to do this on every update? that means every state change...

Reply:
I copied this pattern right from the redesigned more info dialog. How should I be handling this?

Reply:
tabs only change when `_device` changes, so add a `if(!changedProps.has("_device")) return`